### ISSUE TITLE
Strange bindings within bindingHandler using C3 library

### ISSUE THREAD

**Issue by oniseijin** [BladeRunnerJS/brjs#1374]
Similar to another issue about width/height within bindings (perhaps purely KnockoutJS)

Referencing the bottom widget at: http://signaller-rat-84376.bitballoon.com/

Problem: The initial layout has 0 width -> meaning the when binding, it doesn't know the right size. Clicking on the text it displays. However, when rendering purely to a div element (the top of page) it works fine. Unclear where in the toolchain this is getting broken. 

Side problem: C3 doens't resize well within bootstrap, but that is another issue entirely. 

Using the C3 charts library: http://c3js.org/

Code:
```
ko.bindingHandlers.lineChart = {
	init : function(element, valueAccessor){
		c3.generate({
			bindto: element,
			//data: ko.unwrap(valueAccessor()) //not accepted

			data: {
				columns: [
					['data1', 30, 200, 100, 400, 150, 250],
					['data2', 50, 20, 10, 40, 15, 25]
				]
			}

		});
	},

	update: function(element, valueAccessor){

	}
}
```

Template:
```
<div id="kendoui.c3charts.view-template">

	<div id = "c3chart"></div>
	<div data-bind="lineChart: dataset"></div>

</div>
```

Workbench screenshot:
![workbench2](https://cloud.githubusercontent.com/assets/80143/7547509/8f3e6526-f629-11e4-9e07-0806f43fe0ff.png)

**Comment by oniseijin** [comment]
@dchambers  on your advice, this does work, although, would take a bit more work on my end to make the c3 chart reactive to data: 
```
'use strict';

var br = require('br/Core');
var ko = require( 'ko' );
var i18n = require( 'br/I18n' );
var Component = require('br/component/Component');
var c3chart = require("c3");

function C3Component() {
	this.m_oFrame; 
	this._rootElement = document.createElement('div');
	this.getElement().style.width = "300px";
	this.getElement().style.height = "300px";

}

br.extend(C3Component, Component);

C3Component.prototype.getElement = function(){
	return this._rootElement

}

C3Component.prototype.setDisplayFrame = function(frame){
	this.m_oFrame = frame;
	frame.setContent(this.getElement());
	
}

C3Component.prototype.attachChart = function(){
	
	var chart = c3.generate({
		bindto: this.getElement(),
		data: {
			columns: [
				['data1', 30, 200, 100, 400, 150, 250],
				['data2', 50, 20, 10, 40, 15, 25]
			]
		}
	});
	
}


module.exports = C3Component;

```

I then need the following on the workbench: 
```
var C3Component = require( 'experiment/c3/C3Component' );
			//var KnockoutComponent = require( 'br/knockout/KnockoutComponent' );

			var SimpleFrame = require("br/component/SimpleFrame");
			var component = new C3Component();
			var frame = new SimpleFrame(component, null, null);
			document.body.appendChild(frame.getElement());
			component.attachChart()
```

**Comment by dchambers** [comment]
Hi @oniseijin,

If you look at the [`App`](https://github.com/BladeRunnerJS/brjs-todomvc-knockout/blob/master/src/App.js) class of the [Todo MVC](http://todomvc.com/) app built within our [Getting Started Guide](http://bladerunnerjs.org/docs/use/getting_started/) you will notice that the app cheats (it does not fulfil the [`Component`](http://apidocs.bladerunnerjs.org/v1.0-RC1/js/module-br_component_Component.html) contract), and merely adds the element returned via `getElement()` to the page. What's worse, this `getElement()` method is a deprecated method that is only there so that `PresenterComponent` and `KnockoutComponent` continue to be backwardly compatible with the old [`Component`](https://www.caplin.com//assets/static_files/api/caplinjslibraries/3.11.2-292918/#caplin.component.component.html) interface.

Consequently, none of the [`Frame`](http://apidocs.bladerunnerjs.org/v1.0-RC1/js/module-br_component_Frame.html) events like [`attach`](http://apidocs.bladerunnerjs.org/v1.0-RC1/js/module-br_component_Frame.html#event:attach) will be fired. By accident, this happens to be okay because the components built as part of the TODO MVC app don't happen to rely on any of these events, or access to the [`Frame`](http://apidocs.bladerunnerjs.org/v1.0-RC1/js/module-br_component_Frame.html) itself.

However, what's meant to happen is that [`setDisplayFrame()`](http://apidocs.bladerunnerjs.org/v1.0-RC1/js/module-br_component_Component.html#setDisplayFrame) should be invoked, and the component should be provided with a [`Frame`](http://apidocs.bladerunnerjs.org/v1.0-RC1/js/module-br_component_Frame.html) reference. For example:

```js
var C3Component = require( 'experiment/c3/C3Component' );
var SimpleFrame = require("br/component/SimpleFrame");
var frame = new SimpleFrame(new C3Component(), null, null);
document.body.appendChild(frame.getElement());
frame.trigger('attach');
```

where you would listen to the `attach` event like this:

```js
C3Component.prototype.setDisplayFrame = function(frame) {
    frame.setContent(this._rootElement);

    frame.on('attach', function() {
        var chart = c3.generate({
            bindto: this._rootElement,
            data: {
                columns: [
                    ['data1', 30, 200, 100, 400, 150, 250],
                    ['data2', 50, 20, 10, 40, 15, 25]
                ]
            }
        });
    });
};
```

Clearly we've got some work to do since not only is our documentation lacking in this area, but our [Getting Started Guide](http://bladerunnerjs.org/docs/use/getting_started/) guide is encouraging people to use deprecated methods.

@andyberry88, are you happy for us to raise an issue to get this looked at for 1.0?

**Comment by oniseijin** [comment]
Fair enough, also makes it clear to "force" a frame when using KnockoutComponent as well on the fix on the ordering....

**Comment by andyberry88** [comment]
This is fixed by #1379 which will be in the BRJS 1.0 release.
